src/components/Sections/utils.ts

```diff
@@ -118,11 +132,15 @@ export const handleSectionAction = (
   values: { shortcutsLayout: SectionSetting[] },
   save: (newValues: { shortcutsLayout: SectionSetting[] }) => void
 ): void => {
+  const isDisplayMode =
+    displayMode === DisplayMode.DETAILED || displayMode === DisplayMode.COMPACT
   const sectionToSave: SectionSetting = {
     ...section.layout,
     [isMobile ? 'mobile' : 'desktop']: {
       ...section.layout[isMobile ? 'mobile' : 'desktop'],
-      detailedLines: displayMode === DisplayMode.DETAILED
+      ...(isDisplayMode
+        ? { detailedLines: displayMode === DisplayMode.DETAILED }
+        : { grouped: displayMode === GroupMode.GROUPED })
```

Comment:
The naming here is a big confusing. Checking for the variable named `displayMode` to know if we are in `displayMode` was confusing at first. I saw only later that `displayMode` type was not only `DisplayMode` but also `GroupMode`

Maybe renaming the parameter `displayOrGroupMode` would be enough to avoid confusion.

A better solution would be to use a typeguard, but maybe this is too much work for very little benefits, so just changing the name would be enough

Reply:
updated name, was confusing indeed